internal/cache/index/index.go

```diff
@@ -183,25 +183,37 @@ func (idx *Index) UpdateObject(obj *Object) {
 }
 
 // RemoveObject removes an Object's Metadata from the Index
-func (idx *Index) RemoveObject(key string, noLock bool) {
-
-	if !noLock {
-		indexLock.Lock()
-		idx.lastWrite = time.Now()
-	}
+func (idx *Index) RemoveObject(key string) {
+	indexLock.Lock()
+	idx.lastWrite = time.Now()
 	if o, ok := idx.Objects[key]; ok {
 		idx.CacheSize -= o.Size
 		idx.ObjectCount--
-
 		cache.ObserveCacheOperation(idx.name, idx.cacheType, "del", "none", float64(o.Size))
-
 		delete(idx.Objects, key)
 		cache.ObserveCacheSizeChange(idx.name, idx.cacheType, idx.CacheSize, idx.ObjectCount)
 	}
+	indexLock.Unlock()
+}
+
+// RemoveObjects removes a list of Objects' Metadata from the Index
+func (idx *Index) RemoveObjects(keys []string, noLock bool) {
+	if !noLock {
```

Comment:
Question:  It appears RemoveObjects is not used without the lock.   Is this logic needed?

Reply:
It was intentional in case we need to use the function elsewhere in a later release that does not already have a lock on the index.